packages/google_maps_flutter/google_maps_flutter/lib/src/controller.dart

```diff
@@ -24,72 +30,66 @@ class GoogleMapController {
     _GoogleMapState googleMapState,
   ) async {
     assert(id != null);
-    final MethodChannel channel =
-        MethodChannel('plugins.flutter.io/google_maps_$id');
-    await channel.invokeMethod<void>('map#waitForMap');
+    await _googleMapsFlutterPlatform.init(id);
     return GoogleMapController._(
-      channel,
       initialCameraPosition,
       googleMapState,
+      mapId: id,
     );
   }
 
   /// Used to communicate with the native platform.
   ///
   /// Accessible only for testing.
+  // TODO: Remove this once tests are migrated to not need this.
```

Comment:
Can we point to an issue here?

Reply:
Yes. I don't even have an issue yet for this, let me create one.

Reply:
Linked to this issue: https://github.com/flutter/flutter/issues/55504